### PR TITLE
utils(iptables): Factor out a common function for executing commands

### PR DESCRIPTION
**Pull request by vadorovsky**
SaveInto and Restore were almost the same aside from what was executed.

Signed-off-by: Michal Rostecki <vadorovsky@gmail.com>

### CODE REVIEW

File: pkg/utils/iptables.go [hunk 1]
```diff
@@ -27,49 +27,46 @@ func init() {
...
+	path, err := exec.LookPath(cmdName)
 	if err != nil {
 		return err
 	}
 	stderrBuffer := bytes.NewBuffer(nil)
-	args := []string{"iptables-save", "-t", table}
 	cmd := exec.Cmd{
 		Path:   path,
 		Args:   args,
-		Stdout: buffer,
```
Comment (aauren) [inline pkg/utils/iptables.go:41]: The args here needs to contain the command name here as well.

So you'll need to do some work like the following in order for it to work:
```
trueArgs := append([]string{cmdName}, args...)
```

**Review by aauren — CHANGES_REQUESTED** [review]
Thanks for the submission @vadorovsky.

It looks like it needs a small tweak in order to work. When I tested it in my cluster it threw an error:
```
2022-04-05T22:04:36.594653566Z E0405 22:04:36.594568   14423 network_policy_controller.go:253] Aborting sync. Failed to run iptables-save: %!v(MISSING)failed to call iptables-save [-t filter]: exit status 1, (ERROR: No valid subcommand given.                                
2022-04-05T22:04:36.594671003Z Valid subcommands:                                                                                                                                                                                                                                 
2022-04-05T22:04:36.594674132Z  * iptables                                                                                                                                                                                                                                        
2022-04-05T22:04:36.594676686Z  * main4                                                                                                                                                                                                                                           
2022-04-05T22:04:36.594679175Z  * iptables-save                                                                                                                                                                                                                                   
2022-04-05T22:04:36.594681693Z  * save4                                                                                                                                                                                                                                           
2022-04-05T22:04:36.594684203Z  * iptables-restore                                                                                                                                                                                                                                
2022-04-05T22:04:36.594686723Z  * restore4                                                                                                                                                                                                                                        
2022-04-05T22:04:36.594694755Z  * iptables-legacy                                                                                                                                                                                                                                 
2022-04-05T22:04:36.594697534Z  * iptables-legacy-save                                                                                                                                                                                                                            
2022-04-05T22:04:36.594700027Z  * iptables-legacy-restore                                                                                                                                                                                                                         
2022-04-05T22:04:36.594702516Z  * iptables-xml                                                                                                                                                                                                                                    
2022-04-05T22:04:36.594705010Z  * xml                                                                                                                                                                                                                                             
2022-04-05T22:04:36.594707616Z  * ip6tables                                                                                                                                                                                                                                       
2022-04-05T22:04:36.594710110Z  * main6                                                                                                                                                                                                                                           
2022-04-05T22:04:36.594712637Z  * ip6tables-save                                                                                                                                                                                                                                  
2022-04-05T22:04:36.594715142Z  * save6                                                                                                                                                                                                                                           
2022-04-05T22:04:36.594717642Z  * ip6tables-restore                                                                                                                                                                                                                               
2022-04-05T22:04:36.594720150Z  * restore6                                                                                                                                                                                                                                        
2022-04-05T22:04:36.594722634Z  * ip6tables-legacy                                                                                                                                                                                                                                
2022-04-05T22:04:36.594725121Z  * ip6tables-legacy-save                                                                                                                                                                                                                           
2022-04-05T22:04:36.594727598Z  * ip6tables-legacy-restore                                                                                                                                                                                                                        
2022-04-05T22:04:36.594730080Z )
```

Other than that, I'm always in favor of consolidation.

File: pkg/utils/iptables.go (hunk 1 above)
Comment (vadorovsky) [reply pkg/utils/iptables.go:41]: Whoops, sorry for that!. Should be fixed now.

**Comment by vadorovsky** [comment]
Actually it's irrelevant now, #1280 completely changes the iptables code.

**Pull request closed without merging**